### ISSUE TITLE
[Fix Available] java.io.IOException is never thrown in body of corresponding try statement

### ISSUE THREAD

**Issue by Fijxu** [canyonmodded/canyon#62]
When compiling Canyon, for some reason it throws this error:

```
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project canyon-server: Compilation failure
./canyon/Canyon-Server/src/main/java/org/bukkit/craftbukkit/CraftServer.java:[497,11] exception java.io.IOException is never thrown in body of corresponding try statement
```

The error is self explanatory so I just replaced the `IOException` by `Exception` in the line 497 of the `CraftServer.java` file and it compiled successfully with the server working and plugins working.

This can be saved as a `0031-Fix-java.io.IOException-is-never-thrown-in-body.patch` and applied using `git am 0031-Fix-java.io.IOException-is-never-thrown-in-body.patch` inside the `Canyon-Server` folder
```
From 913284e9d0a489bb7f91e06d8a6eeee4567f55f3 Mon Sep 17 00:00:00 2001
From: Fijxu <fijxu@zzls.xyz>
Date: Wed, 4 Oct 2023 17:50:43 -0300
Subject: [PATCH 31/31] Fix java.io.IOException is never thrown in body...

---
 src/main/java/org/bukkit/craftbukkit/CraftServer.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index c1e11efe9..2fc44b5bd 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -494,7 +494,7 @@ public final class CraftServer implements Server {
                 getLogger().info("Converting world '" + name + "'");
                 converter.convert(name, new ConvertProgressUpdater(console));
             }
-        } catch (IOException e) {
+        } catch (Exception e) {
             // launder the exception
             throw new RuntimeException(e);
         }
-- 
2.42.0

```

I would like to make a pull request but I don't really know how to do it since the Canyon-Server is not a Github repository.

I know a lot of people want to download just .jar without any technical knowledge so here is a link of my newly created server.jar: https://files.zzls.xyz/Canyon_Server_b1.7.3/

**Comment by KoboDev** [comment]
I could not reproduce this issue.
I had to even had to install maven, because I havent had it installed on my arch system.
https://pastebin.com/vVHhm0ZC

Although I may have compiled on jdk 21.

Will recheck on jdk8

--------------------------------

I deleted my .m2 folder, set it openjdk-8, and even did the whole process over again.
I cannot see this error.

Both builds, has the server jar run perfectly fine too.
https://pastebin.com/ZPtELWZk

Maybe there is something wrong with my machine or your machine with how its building.
Mind checking your jdk?
